File: Server/Models/Item.cs
public class Item
{
    public string Name { get; set; }
    public string Id { get; set; }
    
    public decimal OriginalPrice { get; set; }
    public decimal? SalePrice { get; set; }
    public bool OnSale { get; set; }

    public string Description { get; set; }
    public int QuantityInStock { get; set; }
    
    public DateTime AvailFrom { get; set; }
    public DateTime AvailTo { get; set; }
    
    public decimal CurrentRating { get; private set; } // Decimal value between 0 and 5

    public Item(
        string name, 
        string id, 
        decimal originalPrice, 
        string description, 
        int quantityInStock, 
        DateTime availFrom, 
        DateTime availTo, 
        decimal? salePrice = null,
        decimal currentRating = 0 // Default rating of 0
    )
    {
        Name = name;
        Id = id;
        OriginalPrice = originalPrice;
        Description = description;
        QuantityInStock = quantityInStock;
        AvailFrom = availFrom;
        AvailTo = availTo;
        SalePrice = salePrice;
        OnSale = salePrice.HasValue && salePrice < originalPrice;
        CurrentRating = ValidateRating(currentRating);
    }

    private decimal ValidateRating(decimal rating)
    {
        return Math.Clamp(rating, 0, 5); // Ensure rating is within 0-5
    }
}

File: WebClient/src/app/models/item.ts
export class Item {
  Name: string;
  Id: string;
  OriginalPrice: number;
  SalePrice?: number;
  OnSale: boolean;
  Description: string;
  QuantityInStock: number;
  AvailFrom: Date;
  AvailTo: Date;
  CurrentRating: number;
  CategoryIds: number[];
  ImageUrl?: string;  // Added property

  constructor(data: {
    Name: string;
    Id: string;
    OriginalPrice: number;
    SalePrice?: number;
    OnSale: boolean;
    Description: string;
    QuantityInStock: number;
    AvailFrom: Date | string;
    AvailTo: Date | string;
    CurrentRating: number;
    CategoryIds: number[];
    ImageUrl?: string; // Optional
  }) {
    this.Name = data.Name;
    this.Id = data.Id;
    this.OriginalPrice = data.OriginalPrice;
    this.SalePrice = data.SalePrice;
    this.OnSale = data.OnSale;
    this.Description = data.Description;
    this.QuantityInStock = data.QuantityInStock;
    this.AvailFrom = new Date(data.AvailFrom);
    this.AvailTo = new Date(data.AvailTo);
    this.CurrentRating = data.CurrentRating;
    this.CategoryIds = data.CategoryIds;
    this.ImageUrl = data.ImageUrl;
  }
}